java/src/test/java/com/pryv/acceptance/ConnectionAccessesTest.java

```diff
@@ -0,0 +1,230 @@
+        assertFalse(apiError);
+        assertTrue(apiSuccess);
+        assertNotNull(accesses);
+        assertTrue(accesses.size() > 0);
+        Boolean found = false;
+        for (Access access: accesses) {
+            if(access.getId().equals(accessId)) {
+                found = true;
+                break;
+            }
+        }
+        assertTrue(found);
+    }
+
+    @Test
+    public void testUpdateAccess() {
+        Access newAccess = new Access();
+        String newAccessName = "newAccess";
+        String newStreamId = "newStreamId";
+        String newId = "testId";
+        Permission.Level newPermissionLevel = Permission.Level.contribute;
+        newAccess.setName(newAccessName);
+        newAccess.addPermission(new Permission(newStreamId, newPermissionLevel, null));
+        newAccess.setId(newId);
+        Long updateTime = System.currentTimeMillis()/1000;
+        connection.accesses.update(accessId, newAccess ,updateCallback);
+        Awaitility.await().until(hasApiResult());
+        assertFalse(apiError);
+        assertTrue(apiSuccess);
+        assertNotNull(access);
+        assertEquals(access.getName(), newAccessName);
+        ArrayList<Permission> permissions = access.getPermissions();
+        assertNotNull(permissions);
+        assertTrue(permissions.size() > 0);
+        Permission firstPermission = permissions.get(0);
+        assertNotNull(firstPermission);
+        assertEquals(firstPermission.getStreamId(), newStreamId);
+        assertEquals(firstPermission.getLevel(), newPermissionLevel);
+        assertEquals(access.getCreated(), creationTime);
+        assertEquals(access.getCreatedBy(), TestCredentials.TOKENID);
+        assertEquals(access.getModifiedBy(), TestCredentials.TOKENID);
+        assertEquals(access.getModified(), updateTime, 180);
+        assertEquals(access.getType(), accessType);
+        assertEquals(access.getToken(), accessToken);
+    }
+
+    private static Callable<Boolean> hasApiResult() {
+        return new Callable<Boolean>() {
+
+            @Override
+            public Boolean call() throws Exception {
+                return (apiSuccess || apiError);
+            }
+        };
+    }
+
+    private static void instantiateCallbacks() {
+        getCallback = new GetCallback() {
+
+            @Override
+            public void onSuccess(String successMessage, List resources, Double serverTime) {
+                accesses = resources;
+                apiSuccess = true;
+            }
+
+            @Override
+            public void onError(String errorMessage, Double serverTime) {
+                logger.log("Error during Access retrieval: " + errorMessage);
+                apiError = true;
+            }
+        };
+        createCallback = new CreateCallback() {
+            @Override
+            public void onSuccess(String successMessage, ApiResource resource, Double serverTime) {
+                access = (Access) resource;
```

Comment:
Can't you avoid casting by specifying the generic type when instanciating the object?

Reply:
This is one of my remaining doubt, I would like to avoid that the developers have to cast a Resource object when using our lib, but I did not find a good solution right now...